services/backup/backup.go

```diff
@@ -0,0 +1,70 @@
+package backup
+
+import (
+	"context"
+	"sync"
+
+	backendconfig "github.com/rudderlabs/rudder-server/config/backend-config"
+)
+
+type StorageSettings map[string]backendconfig.SettingsT
+
+type StorageService interface {
```

Comment:
This looks more like a data retention settings provider, rather than a backup storage service.

IMO, we shouldn't need anything more than the following function in jobsdb.

```
func (jd *HandleT) getBackupFileUploader(ctx context.Context, workspaceID string) (filemanager.FileManager, error)
```

The above function could use the data retention settings provider in order to resolve the latest data retention configuration for this workspace.

Reply:
I agree, this seems tightly coupled to the backend config and this specific use case. It doesn't seem like having it as a lib here in `services` is giving it any reusability. Perhaps it can be simplified a bit and be part of jobsdb instead.

Reply:
I have separated this from jobsdb because storage settings need to be updated from the config backend by subscribing to it.